src/Fantomas/CodePrinter.fs

```diff
@@ -4555,7 +4576,10 @@ and genMemberDefn astContext node =
         !- "inherit "
         +> genType astContext false t
         +> addSpaceBeforeClassConstructor e
-        +> genExpr astContext e
+        +> genExpr
```

Comment:
The code fix should be in this area.
I propose you pattern match on the expression and create a separate case if it `SynExpr.Paren(..)` that is too long.
Use the `expressionFitsOnRestOfLine` to determine if the short or long path should be followed.

Reply:
Hey, many thanks for your notes and suggestions. I will go through them.